climada/engine/unsequa/calc_delta_climate.py

```diff
@@ -0,0 +1,457 @@
+                                                columns = ['aai_agg'])
+        freq_curve_unc_df = pd.DataFrame(freq_curve_list,
+                                    columns=['rp' + str(n) for n in rp])
+        eai_exp_unc_df =  pd.DataFrame(eai_exp_list)
+        # Note: sparse dataframes are not used as they are not nativel y compatible with .to_hdf5
+        at_event_unc_df = pd.DataFrame(at_event_list)
+
+        if calc_eai_exp:
+            exp = self.exp_input_var.evaluate()
+            coord_df = exp.gdf[['latitude', 'longitude']]
+        else:
+            coord_df = pd.DataFrame([])
+
+        return UncImpactOutput(samples_df=samples_df,
+                               unit=unit,
+                               aai_agg_unc_df=aai_agg_unc_df,
+                               freq_curve_unc_df=freq_curve_unc_df,
+                               eai_exp_unc_df=eai_exp_unc_df,
+                               at_event_unc_df=at_event_unc_df,
+                               coord_df=coord_df
+                               )
+
+    def _compute_imp_metrics(self, samples_df, chunksize, processes):
+        """Compute the uncertainty metrics
+
+        Parameters
+        ----------
+        samples_df : pd.DataFrame
+            dataframe of input parameter samples
+        chunksize : int
+            size of the samples chunks
+        processes : int
+            number of processes to use
+
+        Returns
+        -------
+        list
+            values of impact metrics per sample
+        """
+        #Compute impact distributions
+        with log_level(level='ERROR', name_prefix='climada'):
+            p_iterator = _sample_parallel_iterator(
+                samples=samples_df,
+                chunksize=chunksize,
+                exp_initial_input_var=self.exp_initial_input_var,
+                impf_initial_input_var=self.impf_initial_input_var,
+                haz_initial_input_var=self.haz_initial_input_var,
+                exp_final_input_var=self.exp_final_input_var,
+                impf_final_input_var=self.impf_final_input_var,
+                haz_final_input_var=self.haz_final_input_var,
+                rp=self.rp,
+                calc_eai_exp=self.calc_eai_exp,
+                calc_at_event=self.calc_at_event,
+            )
+            if processes > 1:
+                with mp.Pool(processes=processes) as pool:
+                    LOGGER.info('Using %s CPUs.', processes)
+                    imp_metrics = pool.starmap(
+                        _map_impact_calc, p_iterator
+                        )
+            else:
+                imp_metrics = itertools.starmap(
+                    _map_impact_calc, p_iterator
+                    )
+
+        #Perform the actual computation
+        with log_level(level='ERROR', name_prefix='climada'):
+            return _transpose_chunked_data(imp_metrics)
+
+
+def _safe_divide(numerator, denominator, replace_with=np.nan):
```

Comment:
I would consider putting this method into the utils and importing it here. I think this does not belong to this module in particular.

Reply:
done. the method is in util now. including unit tests. I added this aspect to the PR description too.